src/assembly/collections.ts

```diff
@@ -208,7 +264,12 @@ export function upsert(
 
   const texts: string[] = [text];
 
-  const result = hostUpsertToCollection(collection, keys, texts);
+  const labelsArr: string[][] = [];
+  if (labels != null) {
+    labelsArr.push(labels);
+  }
```

Comment:
`labels` can't be null here

```suggestion
  labelsArr.push(labels);
```